quiz/backend/stats/models.py

```diff
@@ -0,0 +1,52 @@
+from django.db import models
+
+
+class TimestampModel(models.Model):
+    """Abstract class for saving timestamps.
+
+    :param created_at: Date and time when given object was created.
+    :type created_at: models.DateTimeField(auto_now_add=True)
+
+    :param updated_at: Date and time of when given object was updated.
+    :type updated_at: models.DateTimeField(auto_now_add=True)
+
+    """
+
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        abstract = True
+
+
+class Badge(TimestampModel):
```

Comment:
Please refer to API diagram, badge model is missing quite a bit of info which will be needed to make it work.
One matter regarding badges is how they are supposed to work. I've assumed that a user may unlock a most basic badge by doing sth (win a game) and then over the course of the game badge may upgrade (win 20/100/1000/10_0000 games and so on)

Reply:
Can you provide the link to the API diagram, because the one in the story seks to be outdated😅

Reply:
https://lucid.app/lucidchart/577f2a81-db5a-40f6-acbd-cede8e4667ea/edit?page=0_0&invitationId=inv_30a54a00-4ba4-47c9-8ca6-3857e15eadd2#

It's still the same diagram

Reply:
I don't understand why should I place fields like `is_obtained`, `obtained_on`, and `progression` in this exact model when it won't correspond with task I'm doing. I can put `badge_url` but other fields I'd leave for profile model relation.😅